<?xml version="1.0"?>

<launch>
  <node pkg="hector_stability_assistance" type="speed_controller_node" name="stability_speed_controller" output="screen" respawn="true">
    <param name="critical_stability_threshold" value="0.5"/>
    <param name="warn_stability_threshold" value="1.0"/>
    <param name="sample_resolution" value="0.05"/>
    <param name="prediction_horizon" value="1.0"/>
    <param name="safety_distance" value="0.15"/>

    <!-- sdf contact estimation lib parameters -->
    <param name="pose_predictor" value="sdf_contact_estimation::SDFContactEstimation"/>
    <rosparam file="$(find hector_stability_assistance)/config/sdf_contact_estimation.yaml" command="load" ns="sdf_pose_predictor"/>
    <rosparam file="$(find sdf_contact_estimation)/config/asterix_shape_config.yaml" command="load" ns="sdf_pose_predictor/shape_model" />
    <param name="sdf_pose_predictor/debug" value="false"/>

    <!-- esdf server parameters -->
    <rosparam file="$(find whole_body_planning_server)/config/esdf_server_common.yaml" command="load" ns="esdf_server"/>
    <param name="esdf_server/update_mesh_every_n_sec" value="0.0"/>
    <remap from="~/esdf_server/esdf_map_in" to="/esdf_server/esdf_map_out" />

    <remap from="~cmd_vel_in" to="/cmd_unthrottled_vel_raw"/>
    <remap from="~cmd_vel_out" to="/cmd_vel_raw"/>
  </node>
</launch>
